GoogleDataTransport/GDTCORLibrary/GDTCORFlatFileStorage.m

```diff
@@ -314,8 +363,12 @@ - (void)removeLibraryDataForKey:(nonnull NSString *)key
   dispatch_async(_storageQueue, ^{
```

Comment:
Since I/O operations are happening on a serial queue (non main queue), can these be a synchronous operation to provide thread safety? This is outside the scope of this PR, but something to be considered for future.

Reply:
This particular method is a part of the API, so it can be called from outside the storage and require explicit synchronization. But I agree, there are couple places where similar methods can be separated to sync unprotected part for internal use and async protected for external use. We already use this approach for methods like [this](https://cs.opensource.google/firebase-sdk/firebase-ios-sdk/+/master:GoogleDataTransport/GDTCORLibrary/GDTCORFlatFileStorage.m;l=503;drc=802cb6a71678084bdff6d7b381cb474eae6b27ef). I would prefer to make such modifications as required.

Reply:
Fair enough to punt on this. can you track this refactor internally (as a bug)?

Reply:
b/168100199